packages/solid-query/src/__tests__/transition.test.tsx

```diff
@@ -0,0 +1,58 @@
+import { fireEvent, render, screen, waitFor } from 'solid-testing-library'
+
+import { createSignal, Show, startTransition, Suspense } from 'solid-js'
+import { createQuery, QueryCache, QueryClientProvider } from '..'
+import { createQueryClient, queryKey, sleep } from './utils'
+
+describe("useQuery's in Suspense mode with transitions", () => {
+  const queryCache = new QueryCache()
+  const queryClient = createQueryClient({ queryCache })
+
+  it('should render the content when the transition is done', async () => {
+    const key = queryKey()
+
+    function Suspended() {
+      const state = createQuery(key, async () => {
+        await sleep(10)
+        return true
+      })
+
+      return <div>{state.data ? 'Message' : 'never'}</div>
```

Comment:
One thing that still breaks (even with the current implementation) is that if you use the `Show` component instead so 

```tsx
<Show when={state.data}>
    <div>Message</div>
</Show>
```

still doesn't work 😭  There's maybe something goofy over here with solid js not resolving the reactivity correctly. I'm trying to create a minimal reproduction of where things are going wrong, but it'll take me more time

Reply:
That's true :( But I believe this must be fixed in `solid-js` core.

This however works:
```js
<div>
  <Show when={state.data}>
    <div>Message</div>
  </Show>
</div>
```

It seems that the conditional component always needs to be wrapped in another one and `Fragment` doesn't help here either.
For this one I'd expect something wrong in the transformations made by `X-plugin-solid` packages.

Reply:
This (along with the `Show`) issue seem to be fixed in `solid-js` already: https://github.com/solidjs/solid/commit/ea2c976add11b20798267c1fa218672189955ef6, though not released yet. Currently my fix breaks the `<Show when={...}` case 😂 , so how should we proceed with this?

I believe that the test case could still be useful.

Reply:
Ah yes! Thanks for creating an issue with Ryan about this. I built the latest version and it seems like I can get it working fine (current implementation) with the new fixes Ryan implemented. I will let solid-js push a new release out to solve this upstream issue. However, the transitions tests would be really useful. I would suggest only adding tests in this pull request for now :D Let me know if you have any questions or concerns!

Reply:
Our CI wont let us publish until all tests pass. So for now can we make it `it.skip`. I'll go back ad remove the skip when we have a new solid-js version out

Reply:
Left only the test & updated the commit message.